### ISSUE TITLE
Related tags update vulnerability

### ISSUE THREAD

**Issue by evazion** [r888888888/danbooru#3835]
Reportbooru makes this API call when it needs to update the related tags for a given tag on Danbooru:

    PUT /related_tag.json?name=tagme&related_tags=<tags>&key=<reportbooru_key>

The `key` is used to prevent unauthorized calls to this endpoint. The check looks like this:

https://github.com/r888888888/danbooru/blob/8833374294638a61e292b72910d7226c47104519/app/controllers/application_controller.rb#L39-L44

The problem is that `Danbooru.config.reportbooru_key` is nil in the default config, so the check can be bypassed simply by omitting the `key` param.

In other words, if you deploy Danbooru in the default configuration it's possible for unauthorized users to tamper with the related tags field.
